app/src/Liquidator.ts

```diff
@@ -211,45 +229,78 @@ export default class Liquidator {
 
   async isSolvent(borrower: string): Promise<boolean> {
     const shortName = borrower.slice(0, 8);
-    try {
-      winston.log(
-        "debug",
-        `Checking solvency of ${shortName} via gas estimation...`
+    winston.log(
+      "debug",
+      `Checking solvency of ${shortName} via gas estimation...`
+    );
+    const estimatedGasResult: EstimateGasLiquidationResult =
+      await this.estimateGasForLiquidation(borrower, Liquidator.MAX_STRAIN);
+    if (estimatedGasResult.success) {
+      return false;
+    } else if (estimatedGasResult.error === LiquidationError.Healthy) {
+      return true;
+    } else if (estimatedGasResult.error === LiquidationError.Grace) {
+      winston.log("info", `⏳ ${shortName} is in grace period`);
+      return true;
+    } else {
+      // Checking the unleashLiquidationTime is a workaround for a none-critical bug.
+      const borrowerContract = new this.web3.eth.Contract(
+        MarginAccountABIJson as AbiItem[],
+        borrower
       );
+      const slot0 = await borrowerContract.methods.slot0().call();
+      const unleashLiquidationTime = slot0.unleashLiquidationTime;
+      if (unleashLiquidationTime === "0") {
+        winston.log(
+          "error",
+          `🚨 ${shortName} is not healthy, but has an unleashLiquidationTime of 0. This is not expected. Error encountered: ${estimatedGasResult.errorMsg}.`
```

Comment:
In this case, all we know is that something weird happened and we're unable to liquidate. They may or may not be healthy.